File: Solution713/Solution713.java
package Solution713;

/*
    713. Subarray Product Less Than K
    Given an array of integers nums and an integer k, 
    return the number of contiguous subarrays where the product of all the elements in the subarray is strictly less than k
    Example 1:

    Input: nums = [10,5,2,6], k = 100
    Output: 8
    Explanation: The 8 subarrays that have product less than 100 are:
    [10], [5], [2], [6], [10, 5], [5, 2], [2, 6], [5, 2, 6]
    Note that [10, 5, 2] is not included as the product of 100 is not strictly less than k.
*/
public class Solution713 {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        if (k <= 1) return 0;
        int count = 0;
        int left = 0;
        int product = 1;

        for (int right = 0; right < nums.length; right++) {
            product *= nums[right];

            while (product >= k) {
                product /= nums[left];
                left++;
            }
            count += (right - left + 1);
        }
        return count;
    }

    public static void main(String[] args) {
        Solution713 sol = new Solution713();
        int[] nums = {10, 5, 2 ,6};
        int k = 100;
        System.out.println("Input: nums = [10, 5, 2, 6], k = " + k);
        int result = sol.numSubarrayProductLessThanK(nums, k);
        System.out.println("Output: " + result);        
    }
}

File: Solution713/Solution713.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution713 {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int target) {
        if (target <= 1) return 0;
        int count = 0;
        int left = 0;
        long long product = 1;

        for (int right = 0; right<nums.size(); right++) {
            product *= nums[right];
            while (product >= target) {
            product /= nums[left];
            left ++;
            }
            count += (right - left + 1);
        }
        return count;
    }
};

int main() {
    Solution713 sol;
    vector<int> nums = {10, 5, 2, 6};
    int target = 100;

    cout << "Input: nums = [10, 5, 2, 6], k = " << target << endl;
    int result = sol.numSubarrayProductLessThanK(nums, target);
    cout << "Output: " << result << endl;

    return 0;
}